### ISSUE TITLE
[Feature Request] Generate Workflow/Pipeline file(s) for CI/CD providers

### ISSUE THREAD

**Issue by NatoNathan** [katoahq/katoa#21]
Katoa offers the ability to right CI/CD pipelines and scrips once and run them in multiple environments such as locally Remote Servers, and CI/CD providers such as GitHub Actions, Gitlab CI, Jenkins, and more.

A nice use ability feature that would help maintain portability, would be the Katoa CLI have a command to automatically generate the required files to run your Katoa pipelines and Jobs in common CI/CD environments.

## Example Usage 

```sh
katoa generate gh-actions # Generate GitHub Actions workflows for each Kotoa pipeline found in the repo

kotoa generate gitlab-ci <job> # Generate a gitlab ci pipeline for a single job
```

Ideally an alias like `g` or `gene would be supported.

## Possible Enhancements

A nice possibility would have an argument like `--secrets-provider` which could accept a secrets provider such as `gh-actions` (the default for GitHub actions workflows), `1Password`, `vault` (by hashicorp) or others. This would then include the relevant configurational required in the workflows to load the secrets and make them available.

Another enhancement would be the ability to configure an environment for a job like 
```gh
kotoa gen gh-actions deploy --environment staging
```

**Comment by endigma** [comment]
I wonder if directly serializing some structs into YAML would be possible instead of templating it, katoa already tracks some basic trigger info for pipelines as well so perhaps it could be expanded into a fat enum for different CI providers so the types make more sense.

**Comment by NatoNathan** [comment]
> I wonder if directly serializing some structs into YAML would be possible instead of templating it

The issue there, is that some CI/CD systems use different languages and syntax like Jenkins uses a Jenkinsfile (Declarative Pipeline). You would probably need to write custom serializers.

**Comment by endigma** [comment]
"Custom serializer" is a road sign pointing directly at text templating, I suppose that's going to be more flexible anyway

**Comment by NatoNathan** [comment]
Yeah, there’s definitely a trade off there, have a more “elegant and safe” solution by using struts and serializers but only can only support a subsets of CI/CD provides that use a a common syntax, or support as many providers and platforms as possible with templating with solutions that probably going to be more prone to bugs and harder to maintain.
